### PR TITLE
remove fields config in input_def

### PR DESCRIPTION
**Pull request by heavengate**
**remove fields config in input_def**

not showing fields config in `input_def` based on following change:
- all data in dictionary format in `dataset -> transform -> dataloader -> model -> output` pipeline
- for `paddle.io.DataLoader` only support numpy array list currently, auto parse key from the first sample and assembly as dictionary for each output
- all operators delete useless key in `__call__` function, e.g. `Gt2YoloTarget` calculate `target` from `gt_bbox, gt_class, gt_score`, delete `gt_class, gt_score` after `target` calculated

**Test**: test `train/eval` on `mask/mask-fpn/faster/cascade/ssd/yolo` ok

### CODE REVIEW

File: ppdet/data/reader.py [hunk 1]
```diff
@@ -34,38 +35,36 @@
 
 
 class Compose(object):
-    def __init__(self, transforms, fields=None, from_=transform,
-                 num_classes=81):
+    def __init__(self, transforms, num_classes=81):
```
Comment (qingqing01) [inline ppdet/data/reader.py:38]: 30行的from . import transform也可以删掉了吧

**Review by qingqing01 — APPROVED** [review]
LGTM

File: configs/_base_/datasets/coco_detection.yml [hunk 2]
```diff
@@ -6,12 +6,14 @@ TrainDataset:
...
     anno_path: annotations/instances_train2017.json
     dataset_dir: dataset/coco
+    data_fields: ['image', 'gt_bbox', 'gt_class', 'is_crowd']
 
 EvalDataset:
   !COCODataSet
     image_dir: val2017
     anno_path: annotations/instances_val2017.json
     dataset_dir: dataset/coco
+    data_fields: ['image']
```
Comment (jerrywgz) [inline configs/_base_/datasets/coco_detection.yml:16]: 如果只是image的话是不是也可以删掉了呢

File: ppdet/data/reader.py (hunk 1 above)
Comment (heavengate) [reply ppdet/data/reader.py:38]: `from . import transform`是43行 `op_cls = getattr(transform, k)`用的，31行 `from .transform import operator, batch_operator`可删除，done

File: configs/_base_/datasets/coco_detection.yml (hunk 2 above)
Comment (heavengate) [reply configs/_base_/datasets/coco_detection.yml]: Done, thanks!

File: configs/_base_/datasets/coco_instance.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+metric: COCO
```
Comment (qingqing01) [inline configs/_base_/datasets/coco_instance.yml:1]: coco_instance和coco_detection的差别是啥

Comment (heavengate) [reply configs/_base_/datasets/coco_instance.yml:1]: instance多带出来`gt_poly`字段，分割的模型预处理会处理这个，检测不会，如果直接默认全带出来的话检测模型没处理`gt_poly`组batch会有问题

**Review by qingqing01 — APPROVED** [review]

File: ppdet/data/source/coco.py [hunk 4]
```diff
@@ -161,6 +164,7 @@ def parse_dataset(self, with_background=True):
             ct += 1
             if self.sample_num > 0 and ct >= self.sample_num:
                 break
+        records = records[:100]
```
Comment (jerrywgz) [inline ppdet/data/source/coco.py:167]: removed？

Comment (heavengate) [reply ppdet/data/source/coco.py]: Done，thanks！

**Review by jerrywgz — APPROVED** [review]
LGTM

**Pull request merged**